sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_constants.py

```diff
@@ -41,7 +41,7 @@
 GEN2_STORAGE_CLIENT_NAME = "Gen2StorageClient"
 DEFAULT_CONNECTION_TIMEOUT = 14400
 STORAGE_URI_REGEX = (
-    r"(https:\/\/([a-zA-Z0-9@:%_\\\-+~#?&=]+)[a-zA-Z0-9@:%._\\\-+~#?&=]+\.?)\/([a-zA-Z0-9@:%._\\\-+~#?&=]+)\/(.*)"
+    r"(https:\/\/([a-zA-Z0-9@:%_\\\-+~#?&=]+)[a-zA-Z0-9@:%._\\\-+~#?&=]+\.?)\/([a-zA-Z0-9@:%._\\\-+~#?&=]+)\/?(.*)"
```

Comment:
What's the reasoning for updating the regex here?

Reply:
The regex expected 4 parts to the URI, it was failing since there was no 4th part. I have updated the regex to make the 4th part as optional.